builder/api/src/main/java/io/helidon/builder/api/Prototype.java

```diff
@@ -0,0 +1,414 @@
+     * <p>
+     * This annotation is only allowed on maps that have a key with a single type parameter,
+     * and value of Object, or with a single type parameter.
+     */
+    @Target(ElementType.METHOD)
+    @Retention(RetentionPolicy.CLASS)
+    @interface SameGeneric {
+    }
+
+    /**
+     * Additional methods from this type will be added to the prototype and its builder.
+     * All methods must be declared as static.
+     * <p>
+     * These methods can be annotated with:
+     * <ul>
+     *     <li>{@link Prototype.FactoryMethod} - to create a static factory method on prototype</li>
+     *     <li>{@link Prototype.BuilderMethod} - to create an additional method on prototype builder, first
+     *     parameter is the builder instance, may have additional parameters, must be void</li>
+     *     <li>{@link Prototype.PrototypeMethod} - to be added to the prototype interface, first parameter
+     *     is the prototype instance, may have additional parameters</li>
+     * </ul>
+     */
+    @Target(ElementType.TYPE)
+    @Retention(RetentionPolicy.CLASS)
+    @interface CustomMethods {
+        /**
+         * Type that implements static methods to be available on the prototype.
+         *
+         * @return type with static methods annotated with {@link Prototype.FactoryMethod}
+         */
+        Class<?> value();
+    }
+
+    /**
+     * Annotated static method of a custom methods type to be added to builder.
+     * First parameter must be the builder type. Return type must be void.
+     */
+    @Target({ElementType.METHOD, ElementType.TYPE})
+    @Retention(RetentionPolicy.CLASS)
+    @interface BuilderMethod {
+    }
+
+    /**
+     * Annotated static method of a custom methods type to be added to prototype interface.
+     * First parameter must be the prototype instance.
+     */
+    @Target({ElementType.METHOD, ElementType.TYPE})
+    @Retention(RetentionPolicy.CLASS)
+    @interface PrototypeMethod {
+    }
+
+    /**
+     * Add additional interfaces to implement by the prototype. Provide correct types (fully qualified) for generics.
+     */
+    @interface Implement {
+        /**
+         * Interfaces to implement, such as {@code java.lang.Comparable<io.helidon.common.types.TypeName>}.
+         *
+         * @return interfaces to implement
+         */
+        String[] value();
+    }
+
+    /**
+     * Mark a getter method as redundant - not important for {@code equals}, {@code hashcode}, and/or {@code toString}.
+     * The generated prototype will ignore the fields for these methods in equals and hashcode methods.
+     * All other fields will be included.
+     * <p>
+     * In case both properties are set to false, it is af this annotation is not present.
+     */
+    @Target(ElementType.METHOD)
+    // note: class retention needed for cases when derived builders are inherited across modules
+    @Retention(RetentionPolicy.CLASS)
+    @interface Redundant {
```

Comment:
I think we need `EqqualityIgnored` and `toStringIgnored` explicitly here, and not try to aggregate it as one thing.  Or maybe `Ignored.Equality` and `Ignored.toString`.

Reply:
We may want that. We can do a follow up for this

Reply:
I did not see this one listed on https://github.com/helidon-io/helidon/issues/7023 either

Reply:
We have agreed on team meeting to keep this open and as is.